storage/namespace.go

```diff
@@ -522,7 +551,23 @@ func (n *dbNamespace) Flush(
 	return res
 }
 
-func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
+func (n *dbNamespace) NeedsFlush(start time.Time, end time.Time) bool {
```

Comment:
The single parameter method would work for flushing, but we need the two degrees of freedom when checking [commit log cleanup times](https://github.com/m3db/m3db/blob/c93ecb1dc6806088f616b5ff8cf76960e9acba90/storage/cleanup.go#L151-L167) 

This way felt less invasive. If you feel strongly I can invert the control flow to make the flush/cleanup manager ask the namespace for the range of times it needs to be flushed.